python/cuml/preprocessing/model_selection.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cudf
+from typing import Union, Tuple
+from numba import jit
+import numpy as np
+
+
+@jit
+def _shuffle_idx(df: cudf.DataFrame) -> np.ndarray:
+    """ Return indices which represent a randomly shuffled version of `df`
+    """
+    return np.random.permutation(len(df))
```

Comment:
could we use cupy here, or eventually the mlprim mentioned in #659 to make this a gpu operation? Even if its not there immediately adding a todo would be good

Reply:
That was actually what I originally wanted, but libcudf wasn't in a place to support that [(ee cudf #1879)](https://github.com/rapidsai/cudf/issues/1879). I'll put in a todo for it